DigitalExcellence/dex-backend
# Feature/77 gdpr remove data

## Description

This pull request adds the feature of deleting userdata so that dex is gdpr compliant.

## Type of change


## Checklist


## Steps to Test or Reproduce
1. Re-seed your database so the placeholder user is created.
2. Create a user.
2. Add a project as that user.
3. Embed the created project
4.  make a DELETE request to the /api/User/{userid}
5. Now if you check the database:
    - The project table entries should be updated, the userId should be 2 (placeholder user)
    - The user table entry should be deleted.
    - The identity user entry should be deleted as well.

## Link to issue

Closes: #77
